crates/types/src/ycbcr422_image.rs

```diff
@@ -54,12 +54,12 @@ impl From<RgbImage> for YCbCr422Image {
                 .into();
                 [left_color, right_color].into()
             })
-            .collect();
+            .collect::<Vec<YCbCr422>>();
 
         Self {
             width_422,
             height,
-            buffer: Arc::new(data),
+            buffer: Arc::from(data),
```

Comment:
How does this behave with respect to copying? Is this just taking ownership and moving the pointers behind `Vec`?